woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -2986,3 +2976,33 @@ def test_nan_index_error(sample_df_pandas):
 
     with pytest.raises(ValueError, match=match):
         nan_df.ww.init_with_full_schema(schema)
+
+
+def test_validate(sample_df):
+    df = sample_df[["email", "age"]]
+    df.ww.init(logical_types={"email": "EmailAddress"})
+    assert df.ww.validate() is None
+
+    invalid_row = pd.Series({4: "bad_email"}, name="email", dtype="string").to_frame()
+
+    if _is_koalas_dataframe(df):
+        invalid_row = ks.from_pandas(invalid_row)
+
+    df = df.append(invalid_row)
+    df.ww.init(logical_types={"email": "EmailAddress"})
+    match = "Series email contains invalid email addresses."
+
+    with pytest.raises(ValueError, match=match):
+        df.ww.validate()
+
+    actual = df.ww.validate(return_invalid_values=True)
+
+    if _is_dask_dataframe(actual):
+        actual = actual.compute()
+
+    if _is_koalas_dataframe(actual):
+        actual = actual.to_pandas()
```

Comment:
Use the `to_pandas` utility function here?

Reply:
updated